app/gui/view/welcome-screen/src/lib.rs

```diff
@@ -0,0 +1,178 @@
+//! WelcomeScreen View.
+//!
+//! It is the opened when the IDE launches without any project or entry point selected. It
+//! displays a list of available projects, template cards and "new project" button.
+//!
+//! The DOM of the view is generated in Rust, while CSS styles for elements are provided in
+//! `app/ide-desktop/lib/content/src/style.css`.
+
+#![warn(missing_docs)]
+
+
+mod side_menu;
+mod template_cards;
+
+use ensogl::prelude::*;
+
+use crate::side_menu::SideMenu;
+use crate::template_cards::TemplateCards;
+
+use enso_frp as frp;
+use ensogl::application;
+use ensogl::application::Application;
+use ensogl::display;
+use ensogl::display::DomSymbol;
+use ensogl::system::web;
+use ensogl::system::web::NodeInserter;
+use std::rc::Rc;
+use wasm_bindgen::closure::Closure;
+use web_sys::MouseEvent;
+
+
+
+// =============
+// === Model ===
+// =============
+
+
+// === ClickClosure ===
+
+/// Type alias for "on-click" event handlers on buttons.
+type ClickClosure = Closure<dyn FnMut(MouseEvent)>;
+
+
+// === Model ===
+
+/// Model of Welcome Screen that generates HTML DOM elements.
+#[derive(Clone, CloneRef, Debug)]
+pub struct Model {
+    application:    Application,
+    logger:         Logger,
+    dom:            DomSymbol,
+    display_object: display::object::Instance,
+    side_menu:      SideMenu,
+    template_cards: TemplateCards,
+}
+
+impl Model {
+    /// Constructor. `frp` is used to set up event handlers on buttons.
+    pub fn new(app: &Application, frp: Frp) -> Self {
+        let application = app.clone_ref();
+        let logger = Logger::new("WelcomeScreen");
+        let display_object = display::object::Instance::new(&logger);
+
+        let side_menu = SideMenu::new(&logger, &frp);
+        let template_cards = TemplateCards::new(&logger, &frp);
+
+        let root = {
+            let root = web::create_div();
+            root.set_class_name("templates-view");
+            root.set_id("templates-view");
+
+            let container = {
+                let container = web::create_div();
+                container.set_class_name("container");
+
+                container.append_or_warn(&side_menu.root, &logger);
+                container.append_or_warn(&template_cards.root, &logger);
+
+                container
+            };
+            root.append_or_panic(&container);
+            root
+        };
+
+        let dom = DomSymbol::new(&root);
+        display_object.add_child(&dom);
+
+        // Use `fullscreen_vis` layer to lock position when panning
+        // TODO: use `welcome_screen` instead
```

Comment:
Left TODO comments. Please remember to fix all TODOS before creating a PR. You can keep TODOS in draft PRs but not in non-draft ones.